marshmallow/schema.py

```diff
@@ -99,6 +101,37 @@ def __new__(mcs, name, bases, attrs):
         for attr in mcs.FUNC_LISTS:
             attr_copy = copy.copy(getattr(klass, attr))
             setattr(klass, attr, attr_copy)
+
+        # Now add in the decorated processors. By doing this after constructing
+        # the class, we let standard inheritance do all the hard work.
+        mro = klass.mro()
+
+        klass.processors = defaultdict(list)
```

Comment:
Makes sense. I don't want to add too many "fix XYZ" mini-commits, so I'll fix that when I revise the PR to also include doing something sensible when the processor does not return anything.